core/src/blocks/database_schema.rs

```diff
@@ -0,0 +1,125 @@
+use crate::blocks::block::{Block, BlockResult, BlockType, Env};
+use crate::Rule;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures::future::try_join_all;
+use pest::iterators::Pair;
+use serde_json::Value;
+use tokio::sync::mpsc::UnboundedSender;
+
+use super::helpers::get_data_source_project;
+
+#[derive(Clone)]
+pub struct DatabaseSchema {}
+
+impl DatabaseSchema {
+    pub fn parse(_block_pair: Pair<Rule>) -> Result<Self> {
+        Ok(DatabaseSchema {})
+    }
+}
+
+#[async_trait]
+impl Block for DatabaseSchema {
+    fn block_type(&self) -> BlockType {
+        BlockType::DatabaseSchema
+    }
+
+    fn inner_hash(&self) -> String {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update("database_schema".as_bytes());
+        format!("{}", hasher.finalize().to_hex())
+    }
+
+    async fn execute(
+        &self,
+        name: &str,
+        env: &Env,
+        _event_sender: Option<UnboundedSender<Value>>,
+    ) -> Result<BlockResult> {
+        let config = env.config.config_for_block(name);
+
+        let err_msg = format!(
+            "Invalid or missing `databases` in configuration for \
+        `database_schema` block `{}` expecting `{{ \"databases\": \
+        [ {{ \"workspace_id\": ..., \"data_source_id\": ..., \"database_id\": ... }}, ... ] }}`",
+            name
+        );
+
+        let databases = match config {
+            Some(v) => match v.get("databases") {
+                Some(Value::Array(a)) => a
+                    .iter()
+                    .map(|v| {
+                        let workspace_id = match v.get("workspace_id") {
+                            Some(Value::String(s)) => s,
+                            _ => Err(anyhow!(err_msg.clone()))?,
+                        };
+                        let data_source_id = match v.get("data_source_id") {
```

Comment:
This is the external data_source_id so the name of the data source here.

Reply:
I don't understand what you mean ? Unless I am mistaken, what we call "data source name" in front is just the `datasource_id` in core (and we never call it data "name" here)